File: src/main/java/com/mercateo/jsonschema/schema/ObjectContext.java
package com.mercateo.jsonschema.schema;

import com.mercateo.immutables.DataClass;
import com.mercateo.immutables.Tuple;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@Tuple
public interface ObjectContext<T> {

    List<T> allowedValues();

    T defaultValue();
}

File: src/main/java/com/mercateo/jsonschema/schema/ObjectContext.kt
package com.mercateo.jsonschema.schema

import com.mercateo.jsonschema.property.Property

data class ObjectContext<T>(
        val property: Property<*, T>,
        val defaultValue: T? = null,
        val allowedValues: List<T> = emptyList()
) {
    val propertyDescriptor get() = property.propertyDescriptor
    val reference get() = property.reference
}
